### ISSUE TITLE
Working on step 7 ("HISAT2 confirmation of removal of human data")

### ISSUE THREAD

**Issue by kmsmith137** [jaleezyy/covid-19-signal#21]
I started working on this

**Comment by fmaguire** [comment]
This was resolved in #7b580ba4e right?

**Comment by jaleezyy** [comment]
I believe we wanted confirmation to be through HISAT2 alignment against the human genome? This looks like the removal of non-SARS-CoV-2 reads, but not necessarily the confirmation step.

**Comment by fmaguire** [comment]
Considering we are throwing away any reads that don't map to the reference with the new "core" (see current branch) we can probably dispense with this entirely.

**Comment by agmcarthur** [comment]
There was discussion in one of the groups that step 5 (removal of non-SARS-CoV-2) be via alignment to the SARS-CoV-2 genome but step 7 verification was alignment against human genome as this combination would absolutely pass ethics/privacy requirements and provide full confidence to our healthcare colleagues.

**Comment by fmaguire** [comment]
Just chatting about that on the cancogen call. Sounds like: map raw sorted reads to human reference and remove any reads that map.  Then go ahead with the remaining reads for trimming.

The human removed but otherwise raw reads can then be uploaded to SRA? 

I can add that and remake the PR.

**Comment by agmcarthur** [comment]
Are we sure removing based on alignment to human won't exclude some legitimate SARS-CoV-2 data? Don't want to create a region of false low coverage.

**Comment by jts** [comment]
I think a useful analysis to inform this discussion would be:

a) take a set of human reads and map them to the ncov reference to see what maps
b) take coronavirus reads and map to human to see what gets lost

**Comment by jts** [comment]
Thinking about it a bit more human WGS won't be representative of the type of off-target sequences we might see so I'm going off the idea of a) a bit. b) is worth doing though

**Comment by robynslee** [comment]
Yeah, not sure about benefit of a. B seems useful, as that's the issue we're worried about. Might be good to try b with a diverse sample of strains from GISAID, to get a sense of this

**Comment by jts** [comment]
Yeah the thought with a) was to test whether mapping to human is necessary to remove any potential host reads. It would be far simpler if we can just map to coronavirus and discard everything else, I wanted to address that with a) but realized it won't really answer the question

**Comment by robynslee** [comment]
Yes, agree, a) doesn't address that question.

**Comment by fmaguire** [comment]
Just throwing this in here so its all together.

- Need to consider mapping scores: https://github.com/galaxyproject/SARS-CoV-2/issues/49
- Could do with a set of SRA archives across the nextstrain tree for doing this analysis (among other QC): e.g. create a composite reference and see if there is a good threshold for distinguishing host contamination.

**Comment by fmaguire** [comment]
I took the one wuhan scheme illumina sample I had to hand and ran BWA-MEM versus a composite human + viral reference.

Of those reads which mapped to viral and human contigs (~200 or 0.02%) the distribution the respective mapping qualities looking like this:  

![composite_reference](https://user-images.githubusercontent.com/1698629/82743633-6c633780-9d44-11ea-979c-8de40dce234c.png)

So most of the small number of problematic reads are a clear viral hit and lower quality human hit (as Torsten suggests in the linked thread). 
If we just take those multihit reads with a MAPQ>=30 to the human reference, we are left with 13 (0.002%) reads:

![multimaps MAPQ>=30](https://user-images.githubusercontent.com/1698629/82743674-fdd2a980-9d44-11ea-9653-7ec811f80baf.png)

We could save a whole 4 of them by comparing the map-qualities between human and viral but honestly based on this dataset I think just using BWA-MEM MAPQ>=30 in the host removal stage is sufficient.
